pkg/processor/sumologicschemaprocessor/aggregate_attributes_processor.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2022 Sumo Logic, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicschemaprocessor
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// aggregateAttributesProcessor
+type aggregateAttributesProcessor struct {
+	aggregations []*aggregation
+}
+
+type aggregation struct {
+	attribute      string
+	patternRegexes []*regexp.Regexp
+}
+
+func newAggregateAttributesProcessor(config []aggregationPair) (*aggregateAttributesProcessor, error) {
+	aggregations := []*aggregation{}
+
+	for i := 0; i < len(config); i++ {
+		pair, err := pairToAggregation(&config[i])
+		if err != nil {
+			return nil, err
+		}
+		aggregations = append(aggregations, pair)
+	}
+
+	return &aggregateAttributesProcessor{aggregations: aggregations}, nil
+}
+
+func pairToAggregation(pair *aggregationPair) (*aggregation, error) {
+	regexes := []*regexp.Regexp{}
+
+	for i := 0; i < len(pair.Patterns); i++ {
+		// We do not support regexes - only wildcards (*). Escape all regex special characters.
+		regexStr := regexp.QuoteMeta(pair.Patterns[i])
+
+		// Replace all wildcards (after escaping they are "\*") with grouped regex wildcard ("(.*)")
+		regexStrWithWildcard := strings.Replace(regexStr, "\\*", "(.*)", -1)
+
+		regex, err := regexp.Compile(regexStrWithWildcard)
+		if err != nil {
+			return nil, err
+		}
+
+		regexes = append(regexes, regex)
+	}
+
+	return &aggregation{attribute: pair.Attribute, patternRegexes: regexes}, nil
+}
```

Comment:
This does not check for exact matches, for example if user specifies pattern `pod_*`, we will match strings like `opod_123`. We should probably discard that, shouldn't we?

Reply:
Yes, we should check for exact matches. Also see my comment here: https://github.com/SumoLogic/sumologic-otel-collector/issues/866#issuecomment-1357672421.